packages/mdx/lib/plugin/recma-jsx-rewrite.js

```diff
@@ -136,11 +136,25 @@ export function recmaJsxRewrite(options = {}) {
             const fullId = ids.join('.')
             const id = name.name
 
+            const isInScope = inScope(currentScope, id)
+
             if (!own.call(fnScope.references, fullId)) {
-              fnScope.references[fullId] = {node, component: true}
+              const parentScope = /** @type {Scope|null} */ (
+                currentScope.parent
+              )
+              const parentNode = parentScope && parentScope.node
+              if (
+                !isInScope ||
+                // If the parent is a _createMdxContent function, it's a top-level component in MDX.
```

Comment:
```suggestion
                // If the parent scope is `_createMdxContent`, then this
                // references a component we can add a check statement for.
```

I think this explains what’s going on a bit better?